set ignored scope widely

sooahleex opened the pull request:
### Changes
set ignored_scope as op_addresses_to_skip too
<!--- What was changed (briefly), how to reproduce (if applicable), what the reviewers should focus on -->

### Reason for changes
pruning consistently tried to apply to ignored scope
<!--- Why should the change be applied -->

### Related tickets

<!--- Post the numerical ID of the ticket, if available -->

### Tests

<!--- How was the correctness of changes tested and whether new tests were added -->

vinnamkim commented on nncf/common/pruning/model_analysis.py:
```diff
@@ -167,6 +167,9 @@ def propagate_can_prune_attr_up(self):
             # Check all output nodes accept_pruned_input attribute
             out_nodes = self.graph.get_next_nodes(node)
+            if out_nodes == []:
+                self.can_prune[node.node_id] =False
```
Could you please auto-format with pylint?
It seems that this package uses https://github.com/openvinotoolkit/nncf/blob/develop/.pylintrc.
Not only for this line `=False`.

daniil-lyakhov commented:
Guys, looks like we were removed from nncf developers group. Please add me as an reviewer and don't merge this without our review!

vinnamkim commented:
> Guys, looks like we were removed from nncf developers group. Please add me as an reviewer and don't merge this without our review!
> 
> cc: @negved

Please request to join at this link: https://github.com/orgs/openvinotoolkit/teams/nncf_pytorch-maintainers/members

vinnamkim commented:
> > Guys, looks like we were removed from nncf developers group. Please add me as an reviewer and don't merge this without our review!
> > cc: @negved
> 
> Please request to join at this link: https://github.com/orgs/openvinotoolkit/teams/nncf_pytorch-maintainers/members

@negvet

daniil-lyakhov commented:
@sooahleex, we already have mechanism for ignoring layers, it's implemented here: https://github.com/openvinotoolkit/nncf/blob/160990cba0473452955c494b2087c4f4160241c7/nncf/common/pruning/node_selector.py#L279-L305

it used at the last step of node selection here: https://github.com/openvinotoolkit/nncf/blob/160990cba0473452955c494b2087c4f4160241c7/nncf/common/pruning/node_selector.py#L218

It's essential to generate masks for each target node even if it's in a ignored scope because of the algorithm implementation. Ignoring layers during mask propagation could lead to inconsistent pruning layout!

daniil-lyakhov commented on nncf/common/pruning/operations.py:
```diff
@@ -204,6 +204,10 @@ def generate_output_mask(cls, node: NNCFNode, graph: NNCFGraph,
+        # if node is leaf node return None
+        if not graph.get_output_edges(node):
+            return None
+
```
Looks like problem is more fundamental: some of the input node has non empty pruning masks. The result of such return could lead to inconsistent pruning layout. @sooahleex, let's try to inspect this graph together

sooahleex commented:
> 



> @sooahleex, we already have mechanism for ignoring layers, it's implemented here:
> 
> https://github.com/openvinotoolkit/nncf/blob/160990cba0473452955c494b2087c4f4160241c7/nncf/common/pruning/node_selector.py#L279-L305
> 
> it used at the last step of node selection here:
> 
> https://github.com/openvinotoolkit/nncf/blob/160990cba0473452955c494b2087c4f4160241c7/nncf/common/pruning/node_selector.py#L218
> 
> It's essential to generate masks for each target node even if it's in a ignored scope because of the algorithm implementation. Ignoring layers during mask propagation could lead to inconsistent pruning layout!

I knew that part and I checked that works out for mask propagation. When I tried to do pruning, the error is occurred in `count_flops_and_weights_per_node` for the layer which I added in ignored scopes. For the function `count_flops_and_weights_per_node`, There is no part to check whether it is prunable. So I added `op_addresses_to_skip` to check whether the layer is prunable or not.
